integration-tests/actions/vrf/vrfv2/setup_steps.go

```diff
@@ -0,0 +1,479 @@
+	commonExistingEnvConfig := testConfig.VRFv2.ExistingEnvConfig.ExistingEnvConfig
+	env, err := test_env.NewCLTestEnvBuilder().
+		WithTestInstance(t).
+		WithTestConfig(&testConfig).
+		WithCustomCleanup(cleanupFn).
+		Build()
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("%s, err: %w", "error creating test env", err)
+	}
+	coordinator, err := env.ContractLoader.LoadVRFCoordinatorV2(*commonExistingEnvConfig.CoordinatorAddress)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("%s, err: %w", "error loading VRFCoordinator2", err)
+	}
+	linkToken, err := env.ContractLoader.LoadLINKToken(*commonExistingEnvConfig.LinkAddress)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("%s, err: %w", "error loading LinkToken", err)
+	}
+	err = vrfcommon.FundNodesIfNeeded(ctx, commonExistingEnvConfig, env.EVMClient, l)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("err: %w", err)
+	}
+	vrfContracts := &vrfcommon.VRFContracts{
+		CoordinatorV2:  coordinator,
+		VRFV2Consumers: nil,
+		LinkToken:      linkToken,
+		BHS:            nil,
+	}
+	vrfKey := &vrfcommon.VRFKeyData{
+		VRFKey:            nil,
+		EncodedProvingKey: [2]*big.Int{},
+		KeyHash:           common.HexToHash(*commonExistingEnvConfig.KeyHash),
+	}
+	return vrfContracts, vrfKey, env, nil
+}
+
+func SetupSubsAndConsumersForExistingEnv(
+	env *test_env.CLClusterTestEnv,
+	coordinator contracts.VRFCoordinatorV2,
+	linkToken contracts.LinkToken,
+	numberOfConsumerContractsToDeployAndAddToSub int,
+	numberOfSubToCreate int,
+	testConfig tc.TestConfig,
+	l zerolog.Logger,
+) ([]uint64, []contracts.VRFv2LoadTestConsumer, error) {
+	var (
+		subIDs    []uint64
+		consumers []contracts.VRFv2LoadTestConsumer
+		err       error
+	)
+	if *testConfig.VRFv2.General.UseExistingEnv {
+		commonExistingEnvConfig := testConfig.VRFv2.ExistingEnvConfig.ExistingEnvConfig
+		if *commonExistingEnvConfig.CreateFundSubsAndAddConsumers {
+			consumers, subIDs, err = SetupNewConsumersAndSubs(
+				env,
+				coordinator,
+				testConfig,
+				linkToken,
+				numberOfConsumerContractsToDeployAndAddToSub,
+				numberOfSubToCreate,
+				l,
+			)
+			if err != nil {
+				return nil, nil, fmt.Errorf("err: %w", err)
+			}
+		} else {
+			consumer, err := env.ContractLoader.LoadVRFv2LoadTestConsumer(*commonExistingEnvConfig.ConsumerAddress)
+			if err != nil {
+				return nil, nil, fmt.Errorf("err: %w", err)
+			}
+			consumers = append(consumers, consumer)
+			subIDs = append(subIDs, *testConfig.VRFv2.ExistingEnvConfig.SubID)
+		}
+	} else {
+		consumers, subIDs, err = SetupNewConsumersAndSubs(
+			env,
+			coordinator,
+			testConfig,
+			linkToken,
+			numberOfConsumerContractsToDeployAndAddToSub,
+			numberOfSubToCreate,
+			l,
+		)
+		if err != nil {
+			return nil, nil, fmt.Errorf("err: %w", err)
+		}
```

Comment:
How about killing most of the if-else nesting by reversing the logic:

```
// just for this case do this...
if *testConfig.VRFv2.General.UseExistingEnv &&  *testConfig.VRFv2.ExistingEnvConfig.ExistingEnvConfig. CreateFundSubsAndAddConsumers {
	consumer, err := env.ContractLoader.LoadVRFv2LoadTestConsumer(*commonExistingEnvConfig.ConsumerAddress)
	if err != nil {
		return nil, nil, fmt.Errorf("err: %w", err)
	}
	consumers = append(consumers, consumer)
	subIDs = append(subIDs, *testConfig.VRFv2.ExistingEnvConfig.SubID)
                     return subIDs, consumers, nil
}

// for all other cases, do this...
consumers, subIDs, err = SetupNewConsumersAndSubs(
	env,
	coordinator,
	testConfig,
	linkToken,
	numberOfConsumerContractsToDeployAndAddToSub,
	numberOfSubToCreate,
	l,
)
if err != nil {
	return nil, nil, fmt.Errorf("err: %w", err)
}
return subIDs, consumers, nil
```

Reply:
yeah, just I usually prefer the most easiest solution